### PR TITLE
I2c documentation

### PR DESCRIPTION
**Pull request by tballmsft**

### CODE REVIEW

**Comment by pelikhan** [comment]
don't reuse branches for PRs, always branch from main for a new PR. or you'll get trouble.

**Comment by pelikhan** [comment]
do you use the i2c output pane?

File: website/docs/developer/i2c.mdx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+title: Communicating over I2C
+sidebar_position: 2.6
+---
+
+# I2C
+
+The I2C class provides basic functions to work with I2C registers and perform raw communication over the I2C bus.
+
+
+# MMA8452Q accelerometer example
```
Comment (pelikhan) [inline website/docs/developer/i2c.mdx:11]: header level 2 here

File: website/docs/developer/i2c.mdx [hunk 2]
```diff
@@ -0,0 +1,51 @@
+---
+title: Communicating over I2C
```
Comment (pelikhan) [inline website/docs/developer/i2c.mdx:2]: keep it short, I2C

File: website/docs/developer/i2c.mdx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+---
+
+# I2C
+
+The I2C class provides basic functions to work with I2C registers and perform raw communication over the I2C bus.
+
+
+# MMA8452Q accelerometer example
+
+```
```
Comment (pelikhan) [inline website/docs/developer/i2c.mdx:13]: make sample as `ts`

File: devs/lib/clientcmds.ts [hunk 4]
```diff
@@ -371,15 +371,57 @@ Array.prototype.reduce = function (callbackfn: any, initialValue: any) {
 
 declare module "@devicescript/core" {
     interface I2C {
+        /**
+         * 
+         * @param devAddr is a 7 bit i2c address
```
Comment (pelikhan) [inline devs/lib/clientcmds.ts:376]: typically `name` and the rest of the sentence may be split so it is likely that the "is " will read funny.

**Comment by pelikhan** [comment]
see comments

**Comment by tballmsft** [comment]
@pelikhan - comments addressed

File: devs/lib/clientcmds.ts [hunk 5]
```diff
@@ -371,15 +371,57 @@ Array.prototype.reduce = function (callbackfn: any, initialValue: any) {
 
 declare module "@devicescript/core" {
     interface I2C {
+        /**
+         *
```
Comment (pelikhan) [inline devs/lib/clientcmds.ts:375]: missing description

**Pull request merged**